ml/dgl_utils.py

```diff
@@ -0,0 +1,280 @@
+
+logger = setup_logger(__name__)
+
+
+def get_vectorizer(name):
+    if type(name) != str:
+        logger.info(f"Returning {type(name)} vectorizer")
+        return name
+    if name == config.DIRTY_CAT:
+        return process_dirty_dataframes
+    if name == config.SKLEARN:
+        # TODO
+        return
+    logging.warning(
+        f"Vectorizer name must be one of [{config.DIRTY_CAT}, {config.SKLEARN}, ..]"
+    )
+    return
+
+
+def get_dataframe_for_target(target, df, name):
+    """
+        Might be doing too much -- general idea is to be able to pass in a target as column and select it from df, while returning a new dataframe with target as column
+        Likewise, if target is already a series, then return a dataframe with column 'name'
+        if target is already a dataframe, send it through without modification
+
+        usage examples:
+            -- target_dataframe = get_dataframe_for_target('some_col', df, '..')
+            -- target_dataframe = get_dataframe_for_target(pd.target_df, df, '..')
+            -- target_dataframe = get_dataframe_for_target(pd.Series, df, column_name)
+
+    :param target: str, series, or dataframe
+    :param df: aux dataframe to use if target is a string
+    :param name: useful for when target is a series, it will become the column name of the returned dataframe
+    :return: dataframe
+    """
+    if type(target) == str:
+        # get the target from the dataframe
+        logger.info(f"Returning {name}-target {target} as DataFrame")
+        return pd.DataFrame({target: df[target].values}, index=df.index)
+    if type(target) == pd.core.series.Series:
+        # use `name` as column header
+        logger.info(f"Returning target as DataFrame with column {name}")
+        return pd.DataFrame({name: target}, index=target.index)
+    if type(target) == pd.core.frame.DataFrame:
+        logger.info(f"Returning {name}-target as itself")
+        return target
+    logger.warning(f"Returning `None` for target")
+    return
+
+
+def check_target_not_in_features(df, y, remove=True):
+    """
+
+    :param df: model dataframe
+    :param y: target dataframe
+    :param remove: whether to remove columns from df, default True
+    :return: dataframes of model and target
+    """
+    if y is None:
+        return df, y
+    remove_cols = []
+    if hasattr(y, "columns"):
+        yc = y.columns
+        xc = df.columns
+        for c in yc:
+            if c in xc:
+                remove_cols.append(c)
+    else:
+        logger.warning(f"Target is not of type(DataFrame) and has no columns")
+    if remove:
+        logger.info(f"Removing {remove_cols} columns from DataFrame")
+        tf = df.drop(columns=remove_cols)
+        return tf, y
+    return df, y
+
+
+def featurize_to_torch(df, y, vectorizer, remove=True):
+    """
+        Vectorize pandas DataFrames of model and target features into Torch compatible tensors
```

Comment:
for `cudf` DFs, looks like a bit of a dance, either:

-- stay on GPU: via dlpack: https://github.com/rapidsai/cudf/issues/5547
-- round trip: cudf -> cpu arrow -> torch

but maybe we wait for cudf until after 1.0